pkg/registry/core/replicationcontroller/strategy.go

```diff
@@ -125,14 +128,31 @@ func (rcStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object)
 func (rcStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
 	controller := obj.(*api.ReplicationController)
 	opts := pod.GetValidationOptionsFromPodTemplate(controller.Spec.Template, nil)
+
+	// Run imperative validation
 	allErrs := corevalidation.ValidateReplicationController(controller, opts)
+
+	// If DeclarativeValidation feature gate is enabled, run both validations
 	if utilfeature.DefaultFeatureGate.Enabled(features.DeclarativeValidation) {
-		declarativeErrs := rest.ValidateDeclaratively(ctx, nil, legacyscheme.Scheme, obj)
+		// Run declarative validation
+		var declarativeErrs field.ErrorList
+		func() {
+			defer func() {
+				if r := recover(); r != nil {
+					declarativeErrs = append(declarativeErrs, field.InternalError(nil, fmt.Errorf("panic during declarative validation: %v", r)))
+				}
+			}()
+			declarativeErrs = ValidateDeclarativelyFunc(ctx, nil, legacyscheme.Scheme, obj)
```

Comment:
Given that we've got this `ValidateDeclarativelyFunc` how about we wrap it with another func, ideally one that lives in a shared package, that does the panic handling?

Reply:
That is, add a `ValidateDeclarativelyRecover` func that takes another `ValidateDeclarativelyFunc` as an arg and runs it and handles any panics.

Reply:
I have added a WithRecover and WithRecoverUpdate method to validate.go and associated tests in validate_test.go.  Now in strategy.go the call looks like:

```go
declarativeErrs := rest.WithRecover(rest.ValidateDeclaratively)(ctx, nil, legacyscheme.Scheme, obj)
```
```go
		declarativeErrs := rest.WithRecover(rest.ValidateDeclaratively)(ctx, nil, legacyscheme.Scheme, obj)
		declarativeErrs = append(
			declarativeErrs,
			rest.WithRecoverUpdate(rest.ValidateUpdateDeclaratively)(ctx, nil, legacyscheme.Scheme, obj, old)...)

```